lib/collection/src/lib.rs

```diff
@@ -411,20 +432,40 @@ impl Collection {
         let limit = limit + 1;
 
         let mut points = Vec::new();
-        for shard in self.all_shards() {
-            let mut shard_points = shard
-                .get()
-                .scroll_by(
-                    segment_searcher,
-                    offset,
-                    limit,
-                    &with_payload_interface,
-                    with_vector,
-                    request.filter.as_ref(),
-                )
-                .await?;
-            points.append(&mut shard_points);
-        }
+
+        match shard_selection {
+            None => {
+                for shard in self.all_shards() {
+                    let mut shard_points = shard
+                        .get()
+                        .scroll_by(
+                            segment_searcher,
+                            offset,
+                            limit,
+                            &with_payload_interface,
+                            with_vector,
+                            request.filter.as_ref(),
+                        )
+                        .await?;
+                    points.append(&mut shard_points);
+                }
+            }
+            Some(shard_selection) => {
+                let local_shard = self.local_shard_by_id(shard_selection)?;
+                let mut shard_points = local_shard
+                    .scroll_by(
+                        segment_searcher,
+                        offset,
+                        limit,
+                        &with_payload_interface,
+                        with_vector,
+                        request.filter.as_ref(),
+                    )
+                    .await?;
```

Comment:
Would it be possible to await all shards simultaneously?

Reply:
Did you mean it for these lines?
```rust
None => {
                for shard in self.all_shards() {
                    let mut shard_points = shard
                        .get()
                        .scroll_by(
                            segment_searcher,
                            offset,
                            limit,
                            &with_payload_interface,
                            with_vector,
                            request.filter.as_ref(),
                        )
                        .await?;
                    points.append(&mut shard_points);
                }
            }
```

Reply:
Ok, those are a little higher than the ones selected

Reply:
I guess we could apply this pattern for all operations which are distributed across shards. (`scroll_by`, `search`, `update_from_client`).

I'd prefer tackle this on in a different PR for more clarity. I could create an issue for later.

Reply:
https://github.com/qdrant/qdrant/issues/467